src/Microsoft.Azure.WebJobs.Host/Singleton/IDistributedLockManager.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Manage distributed lock. A lock is specified by (account, lockId).  This implementation should cooperate with <see cref="SingletonConfiguration"/>
+    /// </summary>
+    /// <remarks>
+    /// The default implementation of this is based on blob leases. 
```

Comment:
I wouldn't reference blobs, or any restrictions here. this is just the interface. Implementation details are owned by implementors.